### PR TITLE
Added new methods for comparing float and double arrays with delta - F…

### LINKED ISSUE #1803: Should I implement assertEquals(float[], float[]) and assertEquals(double[], double[]) versions with delta?
Currently TestNG contains `assertEquals(float[], float[])` and `assertEquals(double[], double[])` overloads, but no versions with `delta` parameter. They are rather useless without them. Should I make a pull request?

### PR DESCRIPTION
**Pull request by atulagrawal**
Fixes #1803

Closes #1803

### Did you remember to?

- [x] Add test case(s)
- [x] Update `CHANGES.txt`

We encourage pull requests that:

* Add new features to TestNG (or)
* Fix bugs in TestNG

If your pull request involves fixing SonarQube issues then we would suggest that you please discuss this with the 
[TestNG-dev](https://groups.google.com/forum/#!forum/testng-dev) before you spend time working on it.

### CODE REVIEW

**Review by krmahadevan — APPROVED** [review]

File: src/main/java/org/testng/Assert.java [hunk 1]
```diff
@@ -379,6 +379,42 @@ public static void assertEquals(float[] actual, float[] expected, String message
...
+  /**
+   * Asserts that two arrays contain the equal elements concerning a delta in the same order. If they do not, an
+   * AssertionError is thrown.
+   *
+   * @param actual the actual value
+   * @param expected the expected value
+   * @param delta the absolute tolerable difference between the actual and expected values
+   */
+  public static void assertEquals(float[] actual, float[] expected, float delta) {
+    assertEquals(actual, expected, delta,"");
```
Comment (juherr) [inline src/main/java/org/testng/Assert.java]: A space is missing after the last `,`

File: src/main/java/org/testng/Assert.java [hunk 2]
```diff
@@ -379,6 +379,42 @@ public static void assertEquals(float[] actual, float[] expected, String message
...
+   * they do not, an
+   * AssertionError is thrown.
+   *
+   * @param actual the actual value
+   * @param expected the expected value
+   * @param delta the absolute tolerable difference between the actual and expected values
+   * @param message the assertion error message
+   */
+  public static void assertEquals(float[] actual, float[] expected, float delta, String message) {
+    if (checkRefEqualityAndLength(actual, expected, message)) return;
```
Comment (juherr) [inline src/main/java/org/testng/Assert.java]: Add brackets

**Review by juherr — CHANGES_REQUESTED** [review]
Just a few remarks about style.
Please check the whole pull request because the 2 remarks are valid at many places.

**Comment by atulagrawal** [comment]
@juherr - I have done the changes. Pls check.

**Review by juherr — APPROVED** [review]
LGTM :+1: 

Ping @cbeust

**Pull request merged**